examples/bandwidthconstraints/src/index.js

```diff
@@ -0,0 +1,214 @@
+'use strict';
+
+var DataSeries = require('../../util/timelinegraph').DataSeries;
+var GraphView = require('../../util/timelinegraph').GraphView;
+var Prism = require('prismjs');
+var Video = require('twilio-video');
+var getRoomCredentials = require('../../util/getroomcredentials');
+var getSnippet = require('../../util/getsnippet');
+var helpers = require('./helpers');
+var waveform = require('../../util/waveform');
+var connectWithBandwidthConstraints = helpers.connectWithBandwidthConstraints;
+var updateBandwidthConstraints = helpers.updateBandwidthConstraints;
+
+var connectOrDisconnect = document.querySelector('input#connectordisconnect');
+var audioBitrateSelector = document.querySelector('select#maxaudiobitrate');
+var audioPreview = document.querySelector('audio#audiopreview');
+var videoBitrateSelector = document.querySelector('select#maxvideobitrate');
+var videoPreview = document.querySelector('video#videopreview');
+var waveformContainer = document.querySelector('div#audiowaveform');
+var roomName = null;
+var room = null;
+var startAudioBitrateGraph = null;
+var startVideoBitrateGraph = null;
+var stopAudioBitrateGraph = null;
+var stopVideoBitrateGraph = null;
+
+/**
+ * Attach the AudioTrack to the HTMLAudioElement and start the Waveform.
+ */
+function attachAudioTrack(track, audioElement) {
+  track.attach(audioElement);
+  waveform.setStream(audioPreview.srcObject);
+  var canvas = waveformContainer.querySelector('canvas');
+  if (!canvas) {
+    waveformContainer.appendChild(waveform.element);
+  }
+}
+
+/**
+ * Attach a Track to one of the HTMLMediaElements and start the bitrate graph.
+ */
+function attachTrack(audioElement, videoElement, starAudioBitrateGraph, startVideoBitrateGraph, track) {
+  if (track.kind === 'audio') {
+    attachAudioTrack(track, audioElement);
+    stopAudioBitrateGraph = starAudioBitrateGraph(1000);
+    return;
+  }
+  track.attach(videoElement);
+  stopVideoBitrateGraph = startVideoBitrateGraph(1000);
+}
+
+/**
+ * Detach the AudioTrack from the HTMLAudioElement and stop the Waveform.
+ */
+function detachAudioTrack(track, audioElement) {
+  track.detach(audioElement);
+  waveform.unsetStream();
+  var canvas = waveformContainer.querySelector('canvas');
+  if (canvas) {
+    canvas.remove();
+  }
+}
+
+/**
+ * Detach a Track from its HTMLMediaElement and stop the bitrate graph.
+ */
+function detachTrack(audioElement, videoElement, track) {
+  if (track.kind === 'audio') {
+    detachAudioTrack(track, audioElement);
+    stopAudioBitrateGraph();
+    return;
+  }
+  track.detach(videoElement);
+  stopVideoBitrateGraph();
+}
+
+/**
+ * Connect to or disconnect from a Room.
+ */
+async function connectToOrDisconnectFromRoom(e) {
+  e.preventDefault();
+  if (room) {
+    e.target.value = 'Connect to Room';
```

Comment:
You already have `connectOrDisconnect`, can you set `connectOrDisconnect.value` instead of `e.target`?